assets/javascripts/modules/validatorFocus.js

```diff
@@ -7,10 +7,26 @@
     }
 
     $('.error-summary a').on('click', function(e) {
+      var $this = $(this);
+      var focusId = $this.attr('data-focuses');
+      if (!focusId)
+      {
+        focusId = ("" + $this.attr("href")).trim();
```

Comment:
$this.attr("href") will not always return string. It might return 'undefined' if attribute doesn't exist.
`$(document.createElement("a")).attr("href")` -> 'undefined'

You're right, `.trim()` is to guard against developer errors but with frameworks like Scala you can't always control the output to the character. Better to be safe than sorry.